tools/custom-checks/bump-openapi-spec-dep-version.ts

```diff
@@ -0,0 +1,161 @@
+  const __dirname = path.dirname(__filename);
+  const SCRIPT_DIR = __dirname;
+  const PROJECT_DIR = path.join(SCRIPT_DIR, "../../");
+  console.log(`${TAG} SCRIPT_DIR=${SCRIPT_DIR}`);
+  console.log(`${TAG} PROJECT_DIR=${PROJECT_DIR}`);
+
+  if (!req) {
+    throw new RuntimeError(`req parameter cannot be falsy.`);
+  }
+  if (!req.argv) {
+    throw new RuntimeError(`req.argv cannot be falsy.`);
+  }
+  if (!req.env) {
+    throw new RuntimeError(`req.env cannot be falsy.`);
+  }
+
+  const globbyOpts: GlobbyOptions = {
+    cwd: PROJECT_DIR,
+    ignore: ["**/node_modules"],
+  };
+
+  const DEFAULT_GLOB = "**/src/main/json/openapi.json";
+  /** cannot be empty */
+  const pkgJsonPaths = await globby(DEFAULT_GLOB, globbyOpts);
+
+  const execPromise = util.promisify(exec);
+  const repoUrl = "https://github.com/hyperledger/cacti";
+  const branchName = "main";
+  const latestVersion = execPromise(
+    `git describe --tags --abbrev=0 ${branchName}`,
+  ).then(({ stdout }) => {
+    const latestTag = stdout.trim().substring(1);
+    console.log(
+      `${TAG} Hyperledger Cacti current version of ${branchName} branch  ${repoUrl} is version ${latestTag}`,
+    );
+    return latestTag;
+  });
+
+  const currentLatestVersion = await latestVersion;
+  const oasPaths = await globby(DEFAULT_GLOB, globbyOpts);
+  console.log(
+    `${TAG} Openapi.json file paths detected (${oasPaths.length}) files`,
+  );
+  console.log(`${TAG} Expected Latest Version: ${currentLatestVersion} \n`);
+  console.log(`${TAG} ---- SCANNING FOR OPENAPI.JSON ---- \n`);
+  console.log(`${TAG} File paths found: `);
+  const errors: string[] = [];
+  const checks = pkgJsonPaths.map(async (pathRel) => {
+    const filePathAbs = path.join(PROJECT_DIR, pathRel);
+    console.log(`${TAG} ${filePathAbs}\n`);
+    const pkgJson: packageJson = await fs.readJSON(filePathAbs);
+    if (typeof pkgJson !== "object") {
+      errors.push(`ERROR: ${pathRel} openapi.json cannot be empty.`);
+      return;
+    }
+    if (!pkgJson) {
+      errors.push(`ERROR: ${pathRel} openapi.json cannot be empty.`);
+      return;
+    }
+    if (!isStdLibRecord(pkgJson)) {
+      return;
+    }
+    const packageVersion = pkgJson.info.version;
+    if (typeof packageVersion == "undefined") {
+      console.log(`${TAG} ---- CHECKING OPENAPI.JSON VERSION ---- \n`);
+      console.log(
+        `${TAG} VERSION CHECK FAILED --------: ${pathRel} the package itself has no "version" field. \n`,
+      );
+    } else if (packageVersion == currentLatestVersion) {
+      console.log(`${TAG} ---- CHECKING OPENAPI.JSON VERSION ---- \n`);
+      console.log(
+        `${TAG} VERSION CHECK PASSED --------: ${pathRel} the package itself correct has version { ${packageVersion} }. \n`,
+      );
+    } else if (packageVersion != currentLatestVersion) {
+      pkgJson.info.version = currentLatestVersion;
+      await fs.writeFile(filePathAbs, JSON.stringify(pkgJson, null, 2));
```

Comment:
@aldousalvarez When I ran the script, it produced a diff that changed the entire file due to whitespace changes. Let's figure out a way to have all the openapi.json's auto-formatted consistently with the same tool (prettier is my recommendation) and then you can apply that formatting after you've saved the JSON files and so the diff will only contain the lines that you actually changed, e.g. the vX.Y.Z declarations.